spec/controllers/list_controller_spec.rb

```diff
@@ -224,28 +224,62 @@
   describe "best_periods_for" do
 
     it "returns yearly for more than 180 days" do
+      SiteSetting.stubs(:top_page_default_timeframe).returns("all")
```

Comment:
You don't need to stub SiteSettings like this. Simply do `SiteSetting.top_page_default_timeframe = 'all` and it'll reset after the test to the default value.